lib/utils.js

```diff
@@ -0,0 +1,355 @@
+  compiledYml.source = paths.folder;
+
+  // source of the DATA used - NATH: this section will change with addition of func to add LIVE/LOCAL data
+  compiledYml.dataSource = options.dataSource;
+
+  // if there is a description, we'll add that to the compiledYml object
+  if (patternObject.description) {
+    compiledYml.description = patternObject.description
+  }
+
+  // if there are includes, we'll add that to the compiledYml object
+  if (patternObject.includes) {
+    compiledYml.includes = patternObject.includes
+  }
+
+  compiledYml.categoryUrl = '';
+  // if there is a category, we'll put this pattern into that category's subfolder
+  if (patternObject.category) {
+
+    // add category to compiledYml
+    compiledYml.category = compiledYml.categoryUrl = patternObject.category
+
+  } else {
+    compiledYml.category = compiledYml.categoryUrl = 'uncategorized';
+  }
+
+  // if there is a category, we'll put this pattern into that category's subfolder
+  if (patternObject.subcategory) {
+
+    // add category to compiledYml
+    compiledYml.subcategory = patternObject.subcategory
+    compiledYml.categoryUrl = path.join(patternObject.category,patternObject.subcategory);
+
+  }
+
+  return compiledYml;
+}
+
+/**
+ * Returns a relative path to a pattern's destination folder
+ * @function getPatternDestPath
+ *
+ */
+var getPatternDestPath = function getPatternDestPath (object, paths, options) {
+
+  return path.join(options.patternImportDest, object.categoryUrl, paths.directory);
+
+}
+
+/**
+ * Creates a vinyl file
+ * @function createFile
+ *
+ */
+var createFile = function createFile (filePath, cwd, base, type) {
+  var contents;
+  //var filePath = path.join(__filename, '..', 'fixtures', filePath);
+
+  if (type == 'stream') {
+    contents = fs.createReadStream(filePath);
+  } else {
+    contents = fs.readFileSync(filePath);
+  }
+
+  return new File({
+    path: filePath,
+    cwd: cwd,
+    base: base,
+    contents: contents
+  });
+};
+
+/**
+ * Writes a variable of content to a file
+ * @function writeFile
+ *
+ */
+var writeFile = function writeFile (dest, contents) {
+
+  fs.outputFile(dest, contents, function (err) {
+    if (err) return console.error(err);
+
+    console.log('File written: '+dest);
+  })
+
+};
+
+/**
+ * Copies a file from one place to another
+ * @function copyFile
+ *
+ */
+var copyFile = function copyFile (src, dest) {
+
+  fs.copy(src, dest, function (err) {
+    if (err) return console.error(err)
+      console.log('File copied from: ' + src + ' to ' + dest);
+  }) // copies file
+
+};
+
+/**
+ * Returns an array of css files used by the pattern and its included patterns
+ * @function getPatternCssFilesArray
+ *
+ */
+var getPatternCssFilesArray = function getPatternCssFilesArray (patternFiles) {
+  var cssArray = [];
+  if(patternFiles.includedFiles !== undefined){
+    if((patternFiles.includedFiles.css !== undefined) && Array.isArray(patternFiles.includedFiles.css)){
```

Comment:
seems like these 2 if statements can be combined:
```js
if(patternFiles.includedFiles !== undefined){
    if((patternFiles.includedFiles.css !== undefined) && Array.isArray(patternFiles.includedFiles.css)){
```
also, ```(patternFiles.includedFiles.css !== undefined)``` is unnecessary as ```Array.isArray(patternFiles.includedFiles.css)``` is here